coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,392 @@
+        """
+        fix_suggestions = """
+        Length issues can be fixed by writing shorter lines of code(splitting
+        long lines into multiple shorter lines); writing shorter files(
+        splitting files into modules, writing shorter methods and classes.).
+        """
+
+
+@Length.subaspect
+class LineLength:
+    """
+    A line's length refers to the number of character found in it.
+    """
+    class docs:
+        example = """
+        print('The length of this line is 38')
+        """
+        example_langague = 'python'
+        importance_reason = """
+        Too long lines make code very difficult to read and maintain.
+        """
+        fix_suggestions = """
+        Splitting long lines of code into multiple shorter lines whenever
+        possible. Avoiding the usage of in-line language specific constructs
+        whenever they results in too long lines.
+        """
+    max_line_length = Taste[int](
+        'Maximum number of character for a line.',
+        (79, 80, 100, 120, 160), default=80)
+
+
+@Length.subaspect
+class FileLength:
+    """
+    A file's length refers to number of character found in it.
+    """
+    class docs:
+        example = """
+        # This file would be a large file if we assume that the max number of
+        # lines per file is 10
+
+        class node:
+            def __init__(self, value, leftmostchild, leftsibling):
+                self.value=value
+                self.leftmostchild=leftmostchild
+                self.leftsibling=leftsibling
+
+        # This is example is just showing what this aspect is about, because
+        # the max number of lines per file is usually 999.
+        """
+        example_language = 'python'
+        importance_reason = """
+        Too long programs (or files) are difficult to read, maintain and
+        understand.
+        """
+        fix_suggestions = """
+        Splitting files into modules, writing shorter methods and classes.
+        """
+    max_file_length = Taste[int](
+        'Maximum number of line for a file',
+        (999,), default=999)
+
+
+@Formatting.subaspect
+class Spacing:
+    """
+    Spacing refers to all whitespace found between non-whitespace characters.
+    """
+    class docs:
+        example = """
+        # Here is an example of code with spacing issues including trailing
+        # spaces(encoded as \\x20 as we don't want them in our code),
+        # optional/useless blank lines and missing space around operators.
+
+
+
+        def func(   ):
+           return.       37*-+2\\x20
+        """
+        example_language = 'python'
+        importance_reason = """
+        Useless spacing affects the readability and maintainability of a code.
+        """
+        fix_suggestions = """
+        Removing the trailing spaces and the meaningless blank lines.
+        """
+
+
+@Spacing.subaspect
+class TrailingSpace:
+    """
+    Trailing space is all whitespace found after the last non-whitespace
```

Comment:
The first line of a docstring should end with a `.` followed by two line feeds. See PEP 257